kernel/signature.ml

```diff
@@ -159,15 +159,15 @@ and add_rule_infos sg (lst:rule_infos list) : unit =
     in
     if infos.stat = Static && !fail_on_symbol_not_found
     then raise (Signature_error (CannotAddRewriteRules (r.l,r.cst)));
-    HId.replace env (id r.cst) {infos with rules = infos.rules @ rs; decision_tree= None}
+    HId.replace env (id r.cst) {infos with rules = rs @ infos.rules; decision_tree= None}
```

Comment:
I think you need to `rev_append` here since rules are meant to be in reverse order. Besides it's faster.

Reply:
You are right, I'll fix that.

Reply:
You are right.